cmd/wallet-sdk-gomobile/credentialschema/credentialschema.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package credentialschema contains functionality for doing credential resolution with credential manifests.
+package credentialschema
+
+import "github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-gomobile/api"
+
+// Resolver can resolve credentials.
+type Resolver struct {
+	credentialReader api.CredentialReader
+}
+
+// NewResolver creates a new Resolver.
+// If a CredentialReader is specified here, then it will be used as the source for VCs in the Resolve method unless a
+// CredentialReader exists in the ResolveOpts.
+func NewResolver(credentialReader api.CredentialReader) *Resolver {
+	return &Resolver{credentialReader: credentialReader}
+}
+
+// ResolveOpts represents the various options for the Resolve method.
+// Only one out of Schema and CredentialsSupported should be used for a given call to Resolve. If both are specified,
+// then Schema will take precedence.
+// At most one out of VCs and CredentialReader should be used for a given call to Resolve. If both are specified,
+// then VCs will take precedence. If neither are specified, then the CredentialReader from the constructor (NewResolver)
+// will be used instead.
+type ResolveOpts struct {
+	// Schema is the schema to use for resolving descriptors.
+	Schema []byte
+	// CredentialsSupported is not yet defined (TODO: define).
+	CredentialsSupported []byte
+	// VCs is a JSON array of Verifiable Credentials. If specified, this takes precedence over the CredentialReader
+	// used in the constructor (NewResolver).
+	VCs []byte
+	// CredentialReader allows for access to a VC storage mechanism. If specified, this takes precedence over the
+	// CredentialReader used in the constructor (NewResolver).
+	CredentialReader api.CredentialReader
+}
+
+// Resolve resolves the given credentials and returns resolved descriptors. See ResolveOpts for more information.
+// Returns a JSON array of resolved descriptors.
+func (c *Resolver) Resolve(resolveOpts ResolveOpts) ([]byte, error) {
```

Comment:
you have to split arguments - 

- schema [manifest OR credentialSuppported]
- Data to resolve - single VC or Credential ID(credential reader has to be present to work with credential ID)

Reply:
Done